IPython/core/interactiveshell.py

```diff
@@ -2044,7 +2044,7 @@ def register_magic_function(self, func, magic_kind='line', magic_name=None):
         self.magics_manager.register_function(func, 
                                   magic_kind=magic_kind, magic_name=magic_name)
 
-    def run_line_magic(self, magic_name, line):
+    def run_line_magic(self, magic_name, line, back_compat=False):
```

Comment:
Maybe make the parameter `_stack_depth=1` rather than `back_compat=False` - that's a bit clearer about what it means. The leading underscore would suggest that it's not for normal use.